src/send/types.ts

```diff
@@ -0,0 +1,364 @@
+  inheritDefault?: boolean;
+  facebook?: {
+    url: string;
+  };
+  instagram?: {
+    url: string;
+  };
+  linkedin?: {
+    url: string;
+  };
+  medium?: {
+    url: string;
+  };
+  twitter?: {
+    url: string;
+  };
+}
+
+
+export interface IBrandSettingsEmail {
+  templateOverride?: IBrandTemplateOverride;
+  head?: {
+    inheritDefault: boolean;
+    content?: string;
+  };
+  footer?: {
+    content?: object;
+    inheritDefault?: boolean;
+    markdown?: string;
+    social?: IBrandSettingsSocialPresence;
+  };
+  header?: {
+    inheritDefault?: boolean;
+    barColor?: string;
+    logo?: {
+      href?: string;
+      image?: string;
+    };
+  };
+}
+
+export interface IBrandSettingsInApp {
+  borderRadius?: string;
+  disableMessageIcon?: boolean;
+  fontFamily?: string;
+  placement?: "top" | "bottom" | "left" | "right";
+  widgetBackground?: {
+    topColor?: string;
+    bottomColor?: string;
+  };
+  colors?: {
+    invertHeader?: boolean;
+    invertButtons?: boolean;
+  };
+  icons?: {
+    bell?: string;
+    message?: string;
+  };
+  preferences?: {
+    templateIds: string[];
+  };
+}
+
+export type IActionButtonStyle = "button" | "link";
+export type IAlignment = "center" | "left" | "right" | "full";
+
+export interface ElementalContent {
+  version: "2022-01-01";
+  brand?: any; // TODO
+  elements: ElementalNode[];
+}
+
+export type ElementalNode =
+  | ElementalTextNode
+  | ElementalMetaNode
+  | ElementalChannelNode
+  | ElementalImageNode
+  | ElementalActionNode
+  | ElementalDividerNode
+  | ElementalGroupNode
+  | ElementalQuoteNode;
+
+export interface ElementalTextNode extends ElementalBaseNode {
+  type: "text";
+  content: string;
+  format?: "markdown";
+  locales?: ElementalLocales;
+}
+
+export interface ElementalMetaNode extends ElementalBaseNode {
+  type: "meta";
+  title?: string;
+}
+
+export interface ElementalChannelNode extends ElementalBaseNode {
+  type: "channel";
+  channel: string;
+  elements?: ElementalNode[];
+  raw?: { [templateName: string]: any };
+}
+
+export interface ElementalImageNode extends ElementalBaseNode {
+  type: "image";
+  src: string;
+  href?: string;
+  align?: IAlignment;
+  altText?: string;
+  width?: string;
+}
+
+export interface ElementalActionNode extends ElementalBaseNode {
+  type: "action";
+  content: string;
+  href: string;
+  actionId?: string;
+  style?: IActionButtonStyle;
+  align?: IAlignment;
+  backgroundColor?: string;
+  locales?: ElementalLocales;
+}
+
+export interface ElementalDividerNode extends ElementalBaseNode {
+  type: "divider";
+  color?: string;
+}
+
+export interface ElementalGroupNode extends ElementalBaseNode {
+  type: "group";
+  elements: ElementalNode[];
+}
+
+export interface ElementalQuoteNode extends ElementalBaseNode {
+  type: "quote";
+  content: string;
+  align?: IAlignment;
+  borderColor?: string;
+  textStyle?: "text" | "h1" | "h2" | "subtext";
+  locales?: ElementalLocales;
```

Comment:
We should avoid exporting the ElementalLocales. It's not exactly spec and right now it is specific to elements with a content property. I'm worried we're asking for a breaking change in the future here.

```suggestion
  locales?: {
    [locale: string]: {
      content: string;
    };
  };
```